src/pybind/mgr/cephadm/module.py

```diff
@@ -3924,6 +3924,48 @@ def remove_osds_status(self) -> List[OSD]:
         """
         return self.to_remove_osds.all_osds()
 
+    @handle_orch_error
+    def set_osd_spec(self, service_name: str, osd_ids: List[str]) -> str:
+        """
+        Update unit.meta file for osd with service name
+        """
+        if service_name not in self.spec_store:
+            raise OrchestratorError(f"Cannot find service '{service_name}' in the inventory")
```

Comment:
we might want to include the fact that users must have applied an OSD service matching the service name they want to attach OSDs to before they can run this